lib/inc/hocon/config_parse_options.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <string>
+#include <memory>
+#include "config_syntax.hpp"
+
+namespace hocon {
+    class config_includer;
```

Comment:
Using class forward-declaration can cut down on the number of include files introduced to each source file. When config_document needs to use config_parse_options, it doesn't necessarily need to know how config_includer is implemented and can avoid including header files related to it.

It's a way to keep compile time down in C++.